src/fpm/generators/occ_grid.py

```diff
@@ -131,23 +132,32 @@ def draw_floorplan_obstacle(g, element, draw, west, south, fill, coords_map, **k
 def draw_floorplan_opening(g, element, draw, west, south, fill, coords_map, **kwargs):
     opening_points = get_opening_points(g, element)
     resolution = kwargs.get("resolution", 0.05)
+    laser_height = kwargs.get("map_laser_height", 0.7)
 
     all_points = list()
     for opening in opening_points:
+        opening_height_max = 0.0
+        opening_height_min = float("inf")
         for face in opening:
-            y_vals = [p.get("y") for p in face]
-            if np.all(np.array(y_vals) == y_vals[0]):
+            z_vals = [p.get("z") for p in face]
+            if not np.all(np.array(z_vals) == z_vals[0]):
```

Comment:
Are these float numbers? Maybe you want to use [`np.allclose`](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html) or sth similar?

Reply:
They should be the same `z` value in this case.

Reply:
Floating points can be sth like `1.000001`, which may cause some unexpected behaviours if you use `==` for comparison.